src/main/java/com/cburch/contracts/BaseMouseListenerContract.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of logisim-evolution.
+ *
+ * Logisim-evolution is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Logisim-evolution is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with logisim-evolution. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Original code by Carl Burch (http://www.cburch.com), 2011.
+ * Subsequent modifications by:
+ *   + College of the Holy Cross
+ *     http://www.holycross.edu
+ *   + Haute École Spécialisée Bernoise/Berner Fachhochschule
+ *     http://www.bfh.ch
+ *   + Haute École du paysage, d'ingénierie et d'architecture de Genève
+ *     http://hepia.hesge.ch/
+ *   + Haute École d'Ingénierie et de Gestion du Canton de Vaud
+ *     http://www.heig-vd.ch/
+ */
+
+package com.cburch.contracts;
+
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+
+/**
+ * Dummy implementation of java.awt.event.MouseListener interface. The main purpose of this
+ * interface is to provide default (empty) implementation of interface methods as, unfortunately
+ * JDKs interfaces do not come with default implementation even they easily could. Implementing this
+ * interface instead of the parent one allows skipping the need of implementing all, even unneeded,
+ * methods. That's saves some efforts and reduces overall LOC.
+ */
+public interface BaseMouseListenerContract extends MouseListener {
+  @Override
+  default void mouseClicked(MouseEvent mouseEvent) {
```

Comment:
IMHO, this call back should be also explicitly overridden by user code.

Reply:
Changed. Still, 7 classes provided no-op implementation.

Reply:
Looks good to me. I agree with the approach and am in favor, as many many files had empty {} just as they needed to be implemented. The only "back side" I see is that for new contributors it is not clear that we use these interfaces instead of the "default" ones. Hence we should clearly document this and make some instructions how to add a "Contract" in the future when required.
As the code base is already quite big, it is easy to lose overview, adding more and more interfaces and classes might help on one side, but might also make more and more confusion. Hence I think that one of the roadmap points should also include a point like "getting into the source code, and where to find what". This might even be a restructuring of the current structure of the code-base.

Reply:
> Hence we should clearly document this and make some instructions how to add a "Contract" in the future when required.

I think this can easily be solved by code style. For internal interfaces we may simply require default implementation whenever possible. The `Contracts` I created were mainly for JDK's classes. But that's also poses no problem because even if any new contributor would use "stock" interfaces instead of Contracts that would be easily noticed once his PR is full of empty implementations. Since there's just a few Contracts, I'd say it's safe to leave it as is and see how it goes. Also we can have simple linter to check that as well.